talpid-types/src/net/wireguard.rs

```diff
@@ -285,6 +285,18 @@ impl From<Box<[u8; 32]>> for PresharedKey {
     }
 }
 
+impl fmt::Debug for PresharedKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", &self)
+    }
+}
+
+impl fmt::Display for PresharedKey {
```

Comment:
I'm not sure we want this type to implement `Display`. The PSK is a secret and we don't want to accidentally print/log/store/show it. It implements `ZeroizeOnDrop` because it should not be spread into more memory/storage than strictly needed.

The `Debug` impl is fine. I think you can modify this PR to have the `Debug` impl print the base64 encoded version and then it's mergable.

Reply:
But Display is implemented for PrivateKey. Or is the PSK treated with even more importance?

Reply:
Given that a PSK is per-peer and not centrally stored it does make sense. Changes made.

Reply:
Oh. With my above argument `PrivateKey` should not implement `Display`. I'm not sure why it does or if it has to for some other reason. But unless it's strictly needed I would prefer avoiding it.